sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -88,7 +88,7 @@ trait PredicateHelper {
    * `false`.
    */
   protected def canEvaluate(expr: Expression, plan: LogicalPlan): Boolean =
-    expr.references.subsetOf(plan.outputSet)
+    !expr.references.isEmpty && expr.references.subsetOf(plan.outputSet)
```

Comment:
This does fix the problem with minimal code changes, but it doesn't match the semantic, as literals CAN be evaluated on any plan.

how about we fix `ExtractEquiJoinKeys` instead?